src/Mod/Draft/DraftVecUtils.py

```diff
@@ -632,7 +636,11 @@ def closest(vector, vlist):
         if d < dist:
             dist = d
             index = i
-    return index
+
+    if returnLength:
+        return (index, dist)
+    else:
+        return index
```

Comment:
This means that in some conditions it will return two items and in other cases it will return only one item. This is inconsistent and means that code that calls this function could break if it doesn't handle the two possibilities.

I think the solution is to always return the same number of values.
```py
if return_length:
    return index, dist
else:
    return index, None
```

However, I think you are doing it like this to keep compatibility with older code. But beware of what I mentioned above.

Also the documentation should say when exactly is the second argument returned.
```
Returns
-------
int
    The index of the list where the closest point is found.

int, float
    The index, and the distance to the closest point, only if `return_length` is `True`.
```

Also, what do you mean `"The distance (no unit)..."`? All distances have units; if no unit is specified it is assumed to be the base FreeCAD unit which is millimeter. Maybe what you mean is the "numerical value".

Reply:
Yes, for backward compatibility purpose.

And the returned distance is just float / numerical value only, no unit like mm.